src/flamenco/runtime/fd_executor.c

```diff
@@ -310,6 +310,88 @@ fd_executor_check_txn_data_sz( fd_exec_txn_ctx_t * txn_ctx ) {
   return FD_RUNTIME_EXECUTE_SUCCESS;
 }
 
+// https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/transaction_processor.rs#L523
+// This function combines the logic of load_program_accounts and load_program_with_pubkey
+int
+fd_executor_check_replenish_program_cache( fd_exec_txn_ctx_t * txn_ctx ) {
+  int err = FD_RUNTIME_EXECUTE_SUCCESS;
+
+  for( ulong i=0UL; i<txn_ctx->accounts_cnt; i++ ) {
+    int hit_max_limit = 0;
+    fd_borrowed_account_t * account = NULL;
+    err                             = fd_txn_borrowed_account_view_idx( txn_ctx, (uchar)i, &account );
+    // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/program_loader.rs#L135
+    if( FD_UNLIKELY( err!=FD_ACC_MGR_SUCCESS ) ) {
+      continue;
+    }
+
+    // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/transaction_processor.rs#L254
+    if( ( !fd_account_is_executable( account->const_meta ) ) ) {
+      continue;
+    }
+    
+    // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/program_loader.rs#L72
+    if( !memcmp( account->const_meta->info.owner, fd_solana_bpf_loader_v4_program_id.uc, sizeof(fd_pubkey_t) ) ) {
+      // ProgramAccountLoadResult::ProgramOfLoaderV4
+      continue;
+    }
+    // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/program_loader.rs#L86
+    if( !memcmp( account->const_meta->info.owner, fd_solana_bpf_loader_deprecated_program_id.uc, sizeof(fd_pubkey_t) ) ) {
+      // ProgramAccountLoadResult::ProgramOfLoaderV1
+      // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/program_loader.rs#L139
+      continue;
+    }
+    // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/program_loader.rs#L90
+    if( !memcmp( account->const_meta->info.owner, fd_solana_bpf_loader_program_id.uc, sizeof(fd_pubkey_t) ) ) {
+      // ProgramAccountLoadResult::ProgramOfLoaderV2
+      // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/program_loader.rs#L150
+      continue;
+    }
+    // https://github.com/anza-xyz/agave/blob/df892c42418047ade3365c1b3ddcf6c45f95d1f1/svm/src/program_loader.rs#L94
+    fd_bpf_upgradeable_loader_state_t program_loader_state = {0};
```

Comment:
nit: code style: make this a pointer `fd_bpf_upgradeable_loader_state_t  program_loader_state[1]`